From 20285ff39f46471677b354a43473d4272fb6e49a Mon Sep 17 00:00:00 2001
From: Vinay Kompella <vinay.kompella@intel.com>
Date: Wed, 7 Mar 2018 12:53:25 +0530
Subject: [PATCH 2/2] Add ArtDefaultGCType as a soong variable.

This change helps us default the GenCopy GC as the default
for broxton

Change-Id: I41de3d9c7143d2d8a7198bb696ff3b60c2db37d3
Tracked-On: https://jira01.devtools.intel.com/browse/CAR-4576
Signed-off-by: Vinay Kompella <vinay.kompella@intel.com>
---
 android/config.go   | 4 ++++
 android/variable.go | 1 +
 2 files changed, 5 insertions(+)

diff --git a/android/config.go b/android/config.go
index dc46381..76affb7 100644
--- a/android/config.go
+++ b/android/config.go
@@ -493,6 +493,10 @@ func (c *config) ArtUseReadBarrier() bool {
 	return Bool(c.ProductVariables.ArtUseReadBarrier)
 }
 
+func (c *config) ArtDefaultGCType() string {
+	return *c.ProductVariables.ArtDefaultGCType;
+}
+
 func (c *deviceConfig) Arches() []Arch {
 	var arches []Arch
 	for _, target := range c.config.Targets[Device] {
diff --git a/android/variable.go b/android/variable.go
index e268a20..36ca337 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -168,6 +168,7 @@ type productVariables struct {
 	SanitizeDeviceArch []string `json:",omitempty"`
 
 	ArtUseReadBarrier *bool `json:",omitempty"`
+	ArtDefaultGCType  *string `json:",omitempty"`
 
 	BtConfigIncludeDir *string `json:",omitempty"`
 
-- 
1.9.1

